assets/app/scripts/services/data.js

```diff
@@ -299,21 +304,21 @@ angular.module('openshiftConsole')
   DataService.prototype.get = function(resource, name, context, opts) {
     resource = APIService.toResourceGroupVersion(resource);
     opts = opts || {};
-
+    var key = this._uniqueKeyForResourceContext(resource, name, context, _.get(opts, 'http.params'));
     var force = !!opts.force;
     delete opts.force;
 
     var deferred = $q.defer();
 
-    var existingData = this._data(resource, context);
+    var existingData = this._data(key);
 
     // If this is a cached resource (immutable resources only), ignore the force parameter
     if (this._isResourceCached(resource) && existingData && existingData.by('metadata.name')[name]) {
       $timeout(function() {
         deferred.resolve(existingData.by('metadata.name')[name]);
       }, 0);
     }
-    else if (!force && this._watchInFlight(resource, context) && this._resourceVersion(resource, context)) {
+    else if (!force && this._watchInFlight(key) && this._resourceVersion(key)) {
```

Comment:
_isResourceCached is whether a particular type of resource should be put
into the cache, i think there was one we didnt want to cache ever

On Fri, Apr 29, 2016 at 3:47 PM, Ben Petersen <notifications@github.com>
wrote:

> In assets/app/scripts/services/data.js
> <https://github.com/openshift/origin/pull/8676#discussion_r61634891>:
>
> >
> >      // If this is a cached resource (immutable resources only), ignore the force parameter
> >      if (this._isResourceCached(resource) && existingData && existingData.by('metadata.name')[name]) {
> >        $timeout(function() {
> >          deferred.resolve(existingData.by('metadata.name')[name]);
> >        }, 0);
> >      }
> > -    else if (!force && this._watchInFlight(resource, context) && this._resourceVersion(resource, context)) {
> > +    else if (!force && this._watchInFlight(key) && this._resourceVersion(key)) {
>
>
>    - add _isCached(key) and ensure it is clearly different than
>    _isResourceCached(resource);
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8676/files/cd51fd4ee136424df96ea874ef68e339fef98098#r61634891>
>